optuna/distributions.py

```diff
@@ -678,3 +678,80 @@ def _get_single_value(distribution: BaseDistribution) -> Union[int, float, Categ
     elif isinstance(distribution, CategoricalDistribution):
         return distribution.choices[0]
     assert False
+
+
+# TODO(himkt): Remove this method with the deletion of deprecated distributions.
+# https://github.com/optuna/optuna/issues/2941
+def _warn_old_distribution_conversion(
+    old_distribution_class_name: str,
+    distribution_class_name: str,
+) -> None:
+    message = (
+        f"{old_distribution_class_name} is deprecated and internally converted to"
+        f" {distribution_class_name}. If you see this message, please update your"
+        f" source code to use {distribution_class_name} instead of"
+        f" {old_distribution_class_name}."
+        " See https://github.com/optuna/optuna/issues/2941."
+    )
```

Comment:
Message reads perfect, exactly what I had in mind! However, do we need to refactor it to separate function? I was thinking we could store the template similarly to [distributions](https://github.com/optuna/optuna/blob/2a208f045cdac0b8c88102fc73f02308f348e493/optuna/distributions.py#L18-L21) and then just format it.

Reply:
> do we need to refactor it to separate function?

Fair enough. I simplify the implementation as you suggested in 825dc5d. The message template is defined inside the method, but I think it's acceptable since it is not used in multiple locations like `_float_distribution_deprecated_msg`. What do you think about it? :eyes:

Reply:
Looks and reads great!